.circleci/config.yml

```diff
@@ -172,19 +172,65 @@ jobs:
             cd /tmp/keep-client/contracts
             gsutil -m cp * gs://${CONTRACT_DATA_BUCKET}/keep-core
 
-  build_token_dashboard_dapp:
+  publish_npm_package:
     executor: docker-node
     steps:
       - attach_workspace:
           at: /tmp/keep-client
-      - setup_remote_docker:
-          docker_layer_caching: true
       - checkout
       - run:
           name: Copy contract artifacts
-          working_directory: ~/project/solidity/dashboard
+          working_directory: ~/project/solidity
           command: |
-            ./scripts/copy-contracts.sh /tmp/keep-client/contracts
+            set -x
+            mkdir -p artifacts
+            cp -r /tmp/keep-client/contracts/* artifacts/
+            name=$(jq --raw-output .name package.json)
+            version=$(jq --raw-output .version package.json)
+            preid=$(echo $version | sed -e s/^.*-\\\([^.]*\\\).*$/\\1/)
+
+            # Find the latest published package version matching this preid.
+            latest_version=$(npm show -json "$name@^$version" version | jq --raw-output ". | sort | .[-1]")
```

Comment:
I think this requires the return to be an `array` or it will fail.  Do we know when there's only a single published package for a particular version, if the return is an `array` or `string`?

Reply:
I think this will return `empty` if we increment the minor version, e.g.  going from `0.12.0-pre` to `0.13.0-pre`, causing the subsequent `latest_version` check to fail.

In the test workflow [here](https://app.circleci.com/pipelines/github/keep-network/keep-core/6089/workflows/33505a29-33df-4c51-be4a-2c332a0d9245/jobs/47320) We're incrementing a minor package version `0.12.0-pre.1` who's minor version matches the result of `version` in this script.

Just want to make sure this behaves as expected when we are publishing `x.x.x-pre.0`

Reply:
That's a great question regarding what `npm` gives us if there's a single result. I would hope still an array, but it seems very likely we'll just get a string.

If there's *no* result, this will leave `latest_version` empty, but we default it in the next line:
```
            latest_version=${latest_version:-$version}
```

This means it will end up being set to `0.13.0-pre`. This is as designed, because when we later to do `npm version prerelease`, it will get bumped to `-pre.0`.

Reply:
< sigh > yep, we get a string when there's a single result, even for `-json`. I know exactly how this decision was made, and I disagree with it so vehemently it ain't even funny lol.

Give me a minute with `jq`.

Reply:
> but we default it in the next line:

Shnikies, totally missed it, thanks for clarifying.

Reply:
Got it. Going to wrap it in an array and then flatten 🙃

Reply:
weeeeeee

Reply:
How did you miss the obvious `${...:-default}` bash syntax?

I can't believe this is how you default things in bash.